src/respend/respendrelayer.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "respend/respendrelayer.h"
+#include "net.h" // RelayTransaction
+#include "primitives/transaction.h"
+#include "protocol.h"
+#include "streams.h"
+#include "util.h"
+#include "utilprocessmsg.h"
+#include <mutex>
+
+namespace respend {
+
+namespace {
+
+// Apply an independent rate limit to double-spend relays
+class RelayLimiter {
+    public:
+        RelayLimiter() : respendCount(0), lastRespendTime(0) { }
+
+        bool HasLimitExceeded(const CTransaction& doubleSpend)
+        {
+            unsigned int size = ::GetSerializeSize(doubleSpend,
+                                                   SER_NETWORK, PROTOCOL_VERSION);
+
+            std::lock_guard<std::mutex> lock(cs);
+            int64_t limit = GetArg("-limitrespendrelay", 100);
+            if (RateLimitExceeded(respendCount, lastRespendTime, limit, size)) {
```

Comment:
you'll need to add a line to allowed_args.cpp.  We don't allow random entries in the bitcoin.conf, has to be an allowed arg.... you can find the same under limitfreerelay in allowed_args.cpp  like this:          .addDebugArg("limitfreerelay=<n>", optionalInt,
            strprintf("Continuously rate-limit free transactions to <n>*1000 bytes per minute (default: %u)",
                         DEFAULT_LIMITFREERELAY))